test/unit/math/mix/fun/lub_constrain_matvar_test.cpp

```diff
@@ -4,20 +4,21 @@ namespace lub_constrain_tests {
 template <typename T1, typename T2, typename T3>
 void expect_matvar(const T1& x, const T2& lb, const T3& ub) {
   auto f1 = [](const auto& x, const auto& lb, const auto& ub) {
-    return stan::math::lub_constrain(x, lb, ub);
+    stan::return_type_t<decltype(x), decltype(lb), decltype(ub)> lp = 0;
+    return stan::math::lub_constrain<false>(x, lb, ub, lp);
   };
   auto f2 = [](const auto& x, const auto& lb, const auto& ub) {
     stan::return_type_t<decltype(x), decltype(lb), decltype(ub)> lp = 0;
-    return stan::math::lub_constrain(x, lb, ub, lp);
+    return stan::math::lub_constrain<true>(x, lb, ub, lp);
   };
   auto f3 = [](const auto& x, const auto& lb, const auto& ub) {
     stan::return_type_t<decltype(x), decltype(lb), decltype(ub)> lp = 0;
-    stan::math::lub_constrain(x, lb, ub, lp);
+    stan::math::lub_constrain<true>(x, lb, ub, lp);
```

Comment:
Isn't this redundant with the test on line 12?  Are we still testing the old `lub_constrain`?  We should be.

Reply:
Yes so for each of these tests <false> corresponds to the constraint without lp and <true> is for the constrain with lp. In order to test the API I thought it would just be easier to change all of these to use it, though eod it still differs to the original functions the test was checking